packages/go_router_builder/lib/src/route_config.dart

```diff
@@ -105,6 +120,39 @@ class RouteConfig {
   final String _path;
   final InterfaceElement _routeDataClass;
   final RouteConfig? _parent;
+  final String? _key;
+  final bool _isShellRoute;
+
+  static String? _decodeKey(InterfaceElement classElement) {
```

Comment:
What is this function trying to grab?

Reply:
This function is grabbing the static getter $navigatorKey. Example:

```dart
@TypedShellRoute<MyShellRouteData>(
  routes: <TypedRoute<RouteData>>[ /* ... */ ],
)
class MyShellRouteData extends ShellRouteData {
  const MyShellRouteData();

  static GlobalKey<NavigatorState>? get $navigatorKey => shellKey; // <- Grabs this.

  @override
  Widget builder(BuildContext context, GoRouterState state, Widget navigator) {
    return MyShellRouteScreen(child: navigator);
  }
}
```